service/history/historyEngine2_test.go

```diff
@@ -212,8 +212,8 @@ func (s *engine2Suite) TestRecordWorkflowTaskStartedSuccessStickyExpired() {
 	expectedResponse := historyservice.RecordWorkflowTaskStartedResponse{}
 	expectedResponse.WorkflowType = msBuilder.GetWorkflowType()
 	executionInfo = msBuilder.GetExecutionInfo()
-	if executionInfo.LastProcessedEvent != common.EmptyEventID {
-		expectedResponse.PreviousStartedEventId = executionInfo.LastProcessedEvent
+	if executionInfo.LastWorkflowTaskCompleteId != common.EmptyEventID {
+		expectedResponse.PreviousStartedEventId = executionInfo.LastWorkflowTaskCompleteId
```

Comment:
is LastProcessedEvent always be WFT completed event? Could it be ActivityTaskCompleted or anything else?
Here what is the PreviousStartedEventId? It does not sound right that you set a "StartedEventId" as WFTCompleted event id.

Reply:
https://github.com/temporalio/temporal/blob/2666ad925b2d627a622b3a0fbb928d5b2bea4572/service/history/workflow/workflow_task_state_machine.go#L744

not sure about the `PreviousStartedEventId`

Reply:
From the line you quote, the LastProcessedEvent is the WorkflowTaskStartedEvent (not the completed event).

Reply:
> From the line you quote, the LastProcessedEvent is the WorkflowTaskStartedEvent (not the completed event).

good catch